### PR TITLE
Bnijmeijer fileio 1 5

### PR DESCRIPTION
**Pull request by BNijmeijer**

### CODE REVIEW

File: MeesGame/Gameplay/GameManagement/GameObject.cs [hunk 1]
```diff
@@ -3,11 +3,11 @@
...
 public abstract class GameObject : IGameLoopObject
 {
-    protected GameObject parent;
-    protected Vector2 position, velocity;
-    protected int layer;
-    protected string id;
-    protected bool visible;
+    public GameObject parent;
+    public Vector2 position, velocity;
+    public int layer;
```
Comment (jensj12) [inline MeesGame/Gameplay/GameManagement/GameObject.cs:8]: Is it still needed to make everything public?

File: MeesGame/Gameplay/Level/FileIO.cs [hunk 2]
```diff
@@ -10,29 +11,34 @@ class FileIO
         /// <summary>
         /// Saves the level to a file with the specified name.
         /// </summary>
-        /// <param name="obj"> Dummy variable. </param>
-        /// <param name="ea"> Dummy variable. </param>
-        public void Save(object obj, EventArgs ea)
+        /// <param name="level"> The level being saved. </param>
+        /// <param name="filename"> The name of the file the level is being saved to. </param>
```
Comment (maxmeijer42) [inline MeesGame/Gameplay/Level/FileIO.cs:15]: Parameter does not exist

File: MeesGame/Gameplay/Level/FileIO.cs [hunk 3]
```diff
@@ -10,29 +11,34 @@ class FileIO
...
+        public static void Save(Level level)
         {
             string fileName = null;
             SaveFileDialog saveFileDialog = new SaveFileDialog();
-            saveFileDialog.Filter = "txt Files (*.txt) | *.txt | All Files (*.*) | *.*";
+            saveFileDialog.Filter = "txt Files (*.txt) | *.txt ";
             if (saveFileDialog.ShowDialog() == DialogResult.OK)
             {
+                fileName = saveFileDialog.FileName;
                 Stream stream = new FileStream(fileName, FileMode.Create);
```
Comment (maxmeijer42) [inline MeesGame/Gameplay/Level/FileIO.cs:24]: Catch exceptions

File: MeesGame/Gameplay/Level/Level.cs [hunk 4]
```diff
@@ -58,7 +59,9 @@ public void ReadXml(XmlReader reader)
 
         public void WriteXml(XmlWriter writer)
```
Comment (maxmeijer42) [inline MeesGame/Gameplay/Level/Level.cs]: This class is never serialized I believe, so it does not need to implement IXmlSerializable

**Review by maxmeijer42 — COMMENTED** [review]
Haven't tested it yet, but it looks good.

File: MeesGame/MeesGame.csproj [hunk 5]
```diff
@@ -47,6 +47,7 @@
     <Compile Include="Gameplay\GameObjects\Tile.cs" />
     <Compile Include="Gameplay\GameObjects\TileField.cs" />
     <Compile Include="Gameplay\GameStates\PlayingLevelState.cs" />
+    <Compile Include="Gameplay\Level\FileIO.cs" />
```
Comment (jensj12) [inline MeesGame/MeesGame.csproj]: FileIO.cs is already in the list. This generates an error.

**Review by jensj12 — CHANGES_REQUESTED** [review]

**Comment by BNijmeijer** [comment]
I've overhauled the FileIO.Save and FileIO.Load methods, because the way I was trying to serialize everything was becoming a huge mess. Now, everything you'd need to create a level is contained in the LevelData class, and as such, both methods use this class in order to (de-)compose levels.

File: MeesGame/Gameplay/Level/FileIO.cs (hunk 2 above)
Comment (BNijmeijer) [reply MeesGame/Gameplay/Level/FileIO.cs:15]: Fixed

File: MeesGame/Gameplay/Level/FileIO.cs (hunk 3 above)
Comment (BNijmeijer) [reply MeesGame/Gameplay/Level/FileIO.cs:24]: Fixed

**Comment by jensj12** [comment]
Merged via #87.

**Pull request closed without merging**